src/CP/core/model.jl

```diff
@@ -192,10 +193,10 @@ end
 
 Check if `model`' statistics are still under the limits.
 """
-belowLimits(model::CPModel) = belowNodeLimit(model) && belowSolutionLimit(model)
+belowLimits(model::CPModel) = belowNodeLimit(model) && belowSolutionLimit(model) && belowTimeLimit(model)
 belowNodeLimit(model::CPModel) = isnothing(model.limit.numberOfNodes) || model.statistics.numberOfNodes < model.limit.numberOfNodes
 belowSolutionLimit(model::CPModel) = isnothing(model.limit.numberOfSolutions) || model.statistics.numberOfSolutions < model.limit.numberOfSolutions
-
+belowTimeLimit(model::CPModel) = isnothing(model.limit.searchingTime) || peektimer() < model.limit.searchingTime

```

Comment:
`peektimer()` can be compared as the loop button on any stopwatch. It gives the elapsed time since the last call to `tic()`.